### PR TITLE
DP-19148 Fix robots on tasks pages

### PR DESCRIPTION
**Pull request by danstafford**
**Description:**
The `tasks` pages are `view_mode_page` pages unlike the `need-to-know`, `resources`, and `related` pages which are `views` pages and also need this specific robots metatag. So on `tasks` pages the URL returned without the `tasks` part and so the tag wasn't being added. I modified this to use the path alias which should do the trick for the use cases here. I added unit tests for this as well.

**Jira:** (Skip unless you are MA staff)
https://jira.mass.gov/browse/DP-19148

**To Test:**
- [ ] These pages should have a robots metatag with "noindex, follow"
http://mass.local/unemployment-insurance-ui-for-employers/tasks
http://mass.local/unemployment-insurance-ui-for-employers/related
http://mass.local/unemployment-insurance-ui-for-employers/resources
http://mass.local/unemployment-insurance-ui-for-employers/need-to-know
- [ ] Check some other pages that should not have this tag

**Screenshots/GIFs:**
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.

---

[Peer Review Checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md)

### CODE REVIEW

File: docroot/modules/custom/mass_metatag/mass_metatag.module [hunk 1]
```diff
@@ -364,16 +364,17 @@ function mass_metatag_page_attachments(array &$attachments) {
     'resources',
   ];
 
-  $current_uri = Url::fromRoute('<current>')->toString();
```
Comment (weitzman) [inline docroot/modules/custom/mass_metatag/mass_metatag.module:367]: I would think that this returns an aliased URL. Thats not what you are seeing?

Comment (danstafford) [reply docroot/modules/custom/mass_metatag/mass_metatag.module:367]: This is the issue I am trying to resolve. If I am on `http://mass.local/unemployment-insurance-ui-for-employers/tasks` then `Url::fromRoute('<current>')->toString();` returns just `/unemployment-insurance-ui-for-employers`. I was looking for a way to determine if this is a `view_mode_page` and get the "tasks" part but I could't figure out how to that.

Comment (weitzman) [reply docroot/modules/custom/mass_metatag/mass_metatag.module:367]: Oh. In that case 'tasks' is the is called an "argument" or a "param". You can get those from some service. The routematch service I think. I'll look around if you tried that already.

Comment (danstafford) [reply docroot/modules/custom/mass_metatag/mass_metatag.module:367]: Yeah I tried `\Drupal::routeMatch()->getParameter('node')` and got the node entity but I can't find the currently used view_mode in there. The view_mode is `all_tasks` which I assume is exclusively used for the `view_mode_page`

Comment (weitzman) [reply docroot/modules/custom/mass_metatag/mass_metatag.module:367]: Is it in -`>getParamaters()`?

Comment (danstafford) [reply docroot/modules/custom/mass_metatag/mass_metatag.module:367]: I tried that too. The only param is the node.

Comment (danstafford) [reply docroot/modules/custom/mass_metatag/mass_metatag.module:367]: Looking at this again I don't think I need to get the alias. I can probably just use `$current_path = \Drupal::service('path.current')->getPath();` . Even though it returns `/unemployment-insurance-ui-for-employers/tasks` for http://mass.local/unemployment-insurance-ui-for-employers/tasks and `/node/7426/related` for http://mass.local/unemployment-insurance-ui-for-employers/related
(/tasks is a view_mode_page and /related is a view-page)

Comment (danstafford) [reply docroot/modules/custom/mass_metatag/mass_metatag.module:367]: OK @weitzman I removed getting the alias like I mentioned in the previous comment.

**Review by weitzman — APPROVED** [review]

**Comment by weitzman** [comment]
Test failure is unrelated

**Pull request merged**